src/azul/indexer/document.py

```diff
@@ -84,7 +84,7 @@ class CataloguedEntityReference(EntityReference):
     catalog: CatalogName
 
     def __str__(self) -> str:
-        return f'{self.catalog}/{super().__str__()}'
+        return f'{self.catalog}/{super(CataloguedEntityReference, self).__str__()}'
```

Comment:
I'm not getting a type error here in PyCharm in 2023.2.3. Besides, the fix looks fishy to me because it reverts back to a deprecated form of the `super()` invocation. Either remove this change or post the warning and your PyCharm version.

Reply:
… or both

Reply:
Found this in my terminal history, however I can now not reproduce. I will remove the change.

<details>
<summary>Terminal log</summary>

```
2023-10-12 17:15:06,603    INFO MainThread azul.indexer.index_service: Writing documents individually
2023-10-12 17:15:06,605    INFO MainThread elasticsearch: Making PUT request to http://127.0.0.1:50607/azul_v2_daniel_test_activities_aggregate/_doc/1509ef40-d1ba-440d-b298-16b7c173dcd4?op_typ
e=create&refresh=true
2023-10-12 17:15:06,606    INFO MainThread elasticsearch: … with request body b'{"entity_id":"1509ef40-d1ba-440d-b298-16b7c173dcd4","contents":{"activities":[{"document_id":"1509ef40-d1ba-440d
-b298-16b7c17...'
2023-10-12 17:15:06,781    INFO MainThread elasticsearch: Got 201 response after 0.175s from PUT to http://127.0.0.1:50607/azul_v2_daniel_test_activities_aggregate/_doc/1509ef40-d1ba-440d-b298
-16b7c173dcd4?op_type=create&refresh=true
2023-10-12 17:15:06,781    INFO MainThread elasticsearch: … with response body '{"_index":"azul_v2_daniel_test_activities_aggregate","_type":"_doc","_id":"1509ef40-d1ba-440d-b298-16b7c173dcd4"
,"_version":1,"…'
--- Logging error ---
Traceback (most recent call last):
  File "/Users/daniel/.pyenv/versions/3.11.5/lib/python3.11/logging/__init__.py", line 1110, in emit
    msg = self.format(record)
          ^^^^^^^^^^^^^^^^^^^
  File "/Users/daniel/.pyenv/versions/3.11.5/lib/python3.11/logging/__init__.py", line 953, in format
    return fmt.format(record)
           ^^^^^^^^^^^^^^^^^^
  File "/Users/daniel/.pyenv/versions/3.11.5/lib/python3.11/logging/__init__.py", line 687, in format
    record.message = record.getMessage()
                     ^^^^^^^^^^^^^^^^^^^
  File "/Users/daniel/.pyenv/versions/3.11.5/lib/python3.11/logging/__init__.py", line 377, in getMessage
    msg = msg % self.args
          ~~~~^~~~~~~~~~~
  File "/Users/daniel/repo/azul1/src/azul/indexer/document.py", line 602, in __str__
    return f'aggregate for {self.entity}'
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daniel/repo/azul1/src/azul/indexer/document.py", line 87, in __str__
    return f'{self.catalog}/{super().__str__()}'
                             ^^^^^^^
TypeError: super(type, obj): obj must be an instance or subtype of type
Call stack:
  File "/Users/daniel/repo/azul1/.venv/bin/coverage", line 8, in <module>
    sys.exit(main())
  File "/Users/daniel/repo/azul1/.venv/lib/python3.11/site-packages/coverage/cmdline.py", line 943, in main
    status = CoverageScript().command_line(argv)
  File "/Users/daniel/repo/azul1/.venv/lib/python3.11/site-packages/coverage/cmdline.py", line 659, in command_line
    return self.do_run(options, args)
  File "/Users/daniel/repo/azul1/.venv/lib/python3.11/site-packages/coverage/cmdline.py", line 830, in do_run
    runner.run()
  File "/Users/daniel/repo/azul1/.venv/lib/python3.11/site-packages/coverage/execfile.py", line 199, in run
    exec(code, main_mod.__dict__)
  File "/Users/daniel/.pyenv/versions/3.11.5/lib/python3.11/unittest/__main__.py", line 18, in <module>
    main(module=None)
  File "/Users/daniel/.pyenv/versions/3.11.5/lib/python3.11/unittest/main.py", line 102, in __init__
    self.runTests()
  File "/Users/daniel/.pyenv/versions/3.11.5/lib/python3.11/unittest/main.py", line 274, in runTests
    self.result = testRunner.run(self.test)
  File "/Users/daniel/.pyenv/versions/3.11.5/lib/python3.11/unittest/runner.py", line 217, in run
    test(result)
  File "/Users/daniel/.pyenv/versions/3.11.5/lib/python3.11/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/Users/daniel/.pyenv/versions/3.11.5/lib/python3.11/unittest/suite.py", line 122, in run
    test(result)
  File "/Users/daniel/.pyenv/versions/3.11.5/lib/python3.11/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/Users/daniel/.pyenv/versions/3.11.5/lib/python3.11/unittest/suite.py", line 122, in run
    test(result)
  File "/Users/daniel/.pyenv/versions/3.11.5/lib/python3.11/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/Users/daniel/.pyenv/versions/3.11.5/lib/python3.11/unittest/suite.py", line 122, in run
    test(result)
  File "/Users/daniel/.pyenv/versions/3.11.5/lib/python3.11/unittest/case.py", line 678, in __call__
    return self.run(*args, **kwds)
  File "/Users/daniel/.pyenv/versions/3.11.5/lib/python3.11/unittest/case.py", line 623, in run
    self._callTestMethod(testMethod)
  File "/Users/daniel/.pyenv/versions/3.11.5/lib/python3.11/unittest/case.py", line 579, in _callTestMethod
    if method() is not None:
  File "/Users/daniel/repo/azul1/test/indexer/test_anvil.py", line 73, in test_indexing
    self._index_canned_bundle(self.bundle)
  File "/Users/daniel/repo/azul1/test/indexer/__init__.py", line 165, in _index_canned_bundle
    cls._index_bundle(bundle, delete=delete)
  File "/Users/daniel/repo/azul1/test/indexer/__init__.py", line 173, in _index_bundle
    cls.index_service.index(cls.catalog, bundle)
  File "/Users/daniel/repo/azul1/src/azul/indexer/index_service.py", line 197, in index
    self.aggregate(tallies)
  File "/Users/daniel/repo/azul1/src/azul/indexer/index_service.py", line 467, in aggregate
    writer.write(new_aggregates)
  File "/Users/daniel/repo/azul1/src/azul/indexer/index_service.py", line 797, in write
    self._write_individually(documents)
  File "/Users/daniel/repo/azul1/src/azul/indexer/index_service.py", line 812, in _write_individually
    self._on_success(doc)
  File "/Users/daniel/repo/azul1/src/azul/indexer/index_service.py", line 861, in _on_success
    log.debug('Successfully wrote %s with %i contribution(s).',
Message: 'Successfully wrote %s with %i contribution(s).'
```

</details>